metagraph/plugins/numpy/types.py

```diff
@@ -367,6 +392,16 @@ def __init__(self, data, mask=None):
     def shape(self):
         return self.value.shape
 
+    def np_matrix(self, copy=False):
```

Comment:
This would be better named `as_dense`. 

Possibly add a parameter "fill_value=0". Change `matrix = self.value * self.mask` to `matrix = self.value.copy(); matrix[self.mask] = fill_value`

Reply:
https://github.com/metagraph-dev/metagraph/pull/133/commits/a443959738386ffe4d2c2ea5801768fcc42ff9fb and https://github.com/metagraph-dev/metagraph/commit/eb0eda1c191b9170b685a88b55e1b3319b42d464 implement this suggestion. 

@jim22k , I used `matrix[~self.mask] = fill_value` instead since it was my understanding that `self.mask` denoted the values that were present and was not a missing value mask. Is this consistent with your understanding?

Reply:
You're right. I keep thinking the mask is the missing values rather than the present values.

To mitigate the situation you describe, can you swap the order of logic checks so the copy is made first (if applicable) before the fill values are updated? That way the call `new_matrix = matrix.as_dense(fill_value=-99, copy=True)` won't change the current matrix. The alternative would be surprising because I asked for a copy.

Reply:
That sounds like a good idea. https://github.com/metagraph-dev/metagraph/pull/133/commits/0c62a0df4b2e9d74a50bb7631097de56fc0bb51f implements this suggestion.